base/cvd/cuttlefish/host/commands/cvd/cli/frontline_parser.cpp

```diff
@@ -17,119 +17,40 @@
 #include "host/commands/cvd/cli/frontline_parser.h"
 
 #include <memory>
+#include <optional>
+#include <string>
 #include <vector>
 
 #include <android-base/file.h>
 #include <android-base/strings.h>
 
 #include "host/commands/cvd/cli/flag.h"
+#include "host/commands/cvd/cli/selector/arguments_separator.h"
 #include "host/commands/cvd/cli/selector/selector_constants.h"
 #include "host/commands/cvd/cli/types.h"
 
 namespace cuttlefish {
 
+using selector::SeparateArguments;
+using selector::SeparatedArguments;
+
 Result<cvd_common::Args> ExtractCvdArgs(cvd_common::Args& args) {
-  FrontlineParser::ParserParam server_param{
-      .server_supported_subcmds = {"*"},
-      .all_args = args
-  };
-  auto frontline_parser = CF_EXPECT(FrontlineParser::Parse(server_param));
-  CF_EXPECT(frontline_parser != nullptr);
+  CF_EXPECT(!args.empty());
+
+  SeparatedArguments separated_arguments = CF_EXPECT(SeparateArguments(args));
 
-  const auto prog_path = frontline_parser->ProgPath();
-  const auto new_sub_cmd = frontline_parser->SubCmd();
-  cvd_common::Args cmd_args{frontline_parser->SubCmdArgs()};
+  cvd_common::Args new_exec_args{separated_arguments.prog_path};
 
-  cvd_common::Args new_exec_args{prog_path};
+  const auto new_sub_cmd = separated_arguments.sub_cmd;
```

Comment:
nit: `s/new_sub_cmd/sub_cmd`. I wasted some time searching for the `old_sub_cmd` here.

Reply:
Done.